src/validation.cpp

```diff
@@ -484,6 +521,15 @@ class MemPoolAccept
         const CTransactionRef& m_ptx;
         const uint256& m_hash;
         TxValidationState m_state;
+        /** A temporary cache containing serialized transaction data for signature verification.
+         * Reused across PolicyScriptChecks and ConsensusScriptChecks. */
+        PrecomputedTransactionData m_precomputed_txdata;
+
+        // Descendant limits passed in to CalculateMemPoolAncestors() for this transaction.
+        // These may be modified while evaluating a transaction (eg to account for in-mempool
+        // conflicts; see the comment in PreChecks).
+        size_t m_limit_descendants;
+        size_t m_limit_descendant_size;
```

Comment:
You're right that it's not necessary yet in #22674 - I apologize, as the cut off for splitting it out of #22290 was a little bit arbitrary, so there are a few not-yet-relevant-until-later refactors in this PR.

To explain the problem: when we add package RBF, we don't want to increase the `MemPoolAccept::m_limit_descendants` here: https://github.com/bitcoin/bitcoin/blob/04670ef81ea2300fcba4e1a492c4c6b0e0752848/src/validation.cpp#L765
For example, if we had a package of 5 parents and 1 child, and each of the parents had exactly 1 conflict, this would cause us to increase the descendant limit 5 times. That would be incorrectly inflating the limit.

Reply:
> To explain the problem: when we add package RBF, we don't want to increase the > MemPoolAccept::m_limit_descendants here:
>
> bitcoin/src/validation.cpp
> 
> Line 765 in 04670ef
> m_limit_descendants += 1; 
>
> For example, if we had a package of 5 parents and 1 child, and each of the parents had exactly 1 > conflict, this would cause us to increase the descendant limit 5 times. That would be incorrectly inflating the limit.

Thanks for the explanation, now I wonder if the carve-out relaxation makes really sense with package RBF. If your counterparty has broadcast a junk branch on the parent transaction and you're RBF'ing one of your output with a package n>2, the one-more-carve-out-child relaxation won't work, I think ? I would say let's defer the conversation for a future PR when package RBF is effectively introduced.